lib/ecto/type.ex

```diff
@@ -863,6 +863,26 @@ defmodule Ecto.Type do
   defp same_date(%Date{} = term), do: {:ok, term}
   defp same_date(_), do: :error
 
+  @doc false
+  @spec filter_empty_values(t, any, [any]) :: {:ok, any} | :empty
+  def filter_empty_values({:array, type}, value, empty_values) when is_list(value) do
+    value = for elem <- value,
+      result = filter_empty_values(type, elem, empty_values),
+      :empty != result,
+      {:ok, elem} = result,
```

Comment:
```suggestion
      {:ok, elem} <- [filter_empty_values(type, elem, empty_values)],
```

Reply:
Thanks, I was wondering if `pattern <- [...]` was idiomatic or not 👍